src/github.com/travis-ci/worker/backend/gce.go

```diff
@@ -644,6 +686,47 @@ func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error)
 	return imagesByName[imageNames[len(imageNames)-1]], nil
 }
 
+func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error) {
+	return p.imageByFilter(fmt.Sprintf(gceImageTravisCIPrefixFilter, language))
+}
+
+func (p *gceProvider) imageSelect(ctx gocontext.Context, startAttributes *StartAttributes) (*compute.Image, error) {
+	p.buildImageSelector()
+
+	imageName, err := p.imageSelector.Select(&image.Params{
+		Infra:    "gce",
+		Language: startAttributes.Language,
+		OsxImage: startAttributes.OsxImage,
+		Dist:     startAttributes.Dist,
+		Group:    startAttributes.Group,
+		OS:       startAttributes.OS,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if imageName == "default" {
+		imageName = p.defaultImage
+	}
+
+	return p.imageByFilter(fmt.Sprintf("name eq ^%s", imageName))
+}
+
+func (p *gceProvider) buildImageSelector() {
+	if p.imageSelector != nil {
+		return
+	}
+
+	switch p.imageSelectorType {
+	case "env":
+		p.imageSelector = image.NewEnvSelector(p.cfg)
+	case "api":
+		baseURL, _ := url.Parse(p.cfg.Get("IMAGE_SELECTOR_URL"))
```

Comment:
This ignores the error that can be returned if the URL can't be parsed. I think it's probably fine to `panic` (similar to `regexp.MustCompile` if it can't be parsed.